Content.Shared/VendingMachines/SharedVendingMachineComponent.cs

```diff
@@ -1,13 +1,17 @@
 using Content.Shared.Actions;
 using Robust.Shared.GameStates;
 using Robust.Shared.Serialization;
+using Robust.Shared.Serialization.TypeSerializers.Implementations.Custom.Prototype;
 
 namespace Content.Shared.VendingMachines
 {
     [Virtual]
     [NetworkedComponent()]
     public class SharedVendingMachineComponent : Component
```

Comment:
Should VendingMachineComponent be a separate client component as well and make this abstract>

Reply:
Fixed. 

I have NFI why it was even like that, there is also functionality for the client to have information about vending machine inventories to reduce pop-in when opening the menu, So I also fixed that by moving some server logic to an abstract shared system.